kafka-0.10.x/src/main/scala/monix/kafka/Deserializer.scala

```diff
@@ -26,17 +26,33 @@ import language.existentials
 /** Wraps a Kafka `Deserializer`, provided for
   * convenience, since it can be implicitly fetched
   * from the context.
+  *
+  * @param className is the full package path to the Kafka `Deserializer`
+  *
+  * @param classType is the actual [[Class]] for [[className]]
+  *
+  * @param constructor creates an instance of [[classType]].
+  *        This is defaulted with a `Deserializer.Constructor[A]` function that creates a
+  *        new instance using an assumed empty constructor.
+  *        Supplying this parameter allows for manual provision of the `Deserializer`.
   */
 final case class Deserializer[A](
   className: String,
-  classType: Class[_ <: KafkaDeserializer[A]]) {
+  classType: Class[_ <: KafkaDeserializer[A]],
+  constructor: Deserializer.Constructor[A] = (d: Deserializer[A]) => d.classType.newInstance()) {
```

Comment:
@alexandru unfortunately, I couldn't simplify this to `_.classType.newInstance()`. There is a 

> missing parameter type for expanded function 

warning, but because `-Xfatal-warnings` is enabled, this makes it impossible to compile.